### PR TITLE
feat(docs): generate single-page doc from wiki

### PR DESCRIPTION
**Pull request by sidvishnoi**
Plan is to generate (using respec2html) a `~/static/docs/index.html` file on a schedule (or manually), which will be served at https://respec.org/docs
The "live" version can be accessed at https://respec.org/docs/src.html (I checked in this branch on server)

Very raw right now, but also pretty solid :smile: 
Some of the changes in this, mainly markdown processing can be used upsteam also.

### CODE REVIEW

File: static/docs/respecConfig.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+function cleanup() {
+  const attributesToRemove = [
+    "data-max-toc",
+    "data-include-name",
+    "data-oninclude",
+  ];
+  for (const attr of attributesToRemove) {
+    document
+      .querySelectorAll(`[${attr}]`)
```
Comment (saschanaz) [inline static/docs/respecConfig.js:184]: Maybe this could be a single query?

Comment (sidvishnoi) [reply static/docs/respecConfig.js:184]: Heh. This one is not yet ready for review. I meant the other one where I linked it.

File: static/docs/respecConfig.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+/**
+ * @param {*} _
+ * @param {string} content
+ */
+function fixMarkupOnInclude(_, content) {
+  const ZERO_WIDTH_SPACE = "&#8203;";
+  // Escape [[[foo]]] and [[foo]] by adding zero-width space. We remove these
+  // extraneous characters in postProcess, so users don't end up copy pasting
+  // them.Ugly, but  other way is upstream changes for an extreme edge case.
```
Comment (marcoscaceres) [inline static/docs/respecConfig.js:71]:
```suggestion
  // them. Ugly, but  other way is upstream changes for an extreme edge case.
```

File: static/docs/respecConfig.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+function fixLinks() {
+  const urlBase = location.origin + "/docs/";
+  /** @type {NodeListOf<HTMLAnchorElement>} */
+  const anchors = document.querySelectorAll(
+    "section a:not([href^=http]):not([href^='mailto:']):not([href^='#'])",
+  );
+  for (const a of anchors) {
+    const href = a.href.split(urlBase, 2)[1].split("#").pop();
```
Comment (marcoscaceres) [inline static/docs/respecConfig.js:126]: could this be done with `new URL()`?

File: static/docs/respecConfig.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    section.querySelector("h2, h3, h4, h5, h6").append(wikiLink);
+  }
+}
+
+function cleanup() {
+  const attributesToRemove = [
+    "data-max-toc",
+    "data-include-name",
+    "data-oninclude",
+    "aria-busy",
```
Comment (marcoscaceres) [inline static/docs/respecConfig.js:168]: maybe don't need this one anymore?

File: static/docs/src.html [hunk 5]
```diff
@@ -0,0 +1,193 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>
+      ReSpec Documentation
+    </title>
+    <script src=
+    "https://deploy-preview-3007--respec-pr.netlify.app/respec-w3c.js" class=
```
Comment (marcoscaceres) [inline static/docs/src.html:9]: half joking... but maybe we need a new documentation profile..

Comment (sidvishnoi) [reply static/docs/src.html:9]: w3c one should be fine I think.. though we don't use many of its features, and maintaining a new profile would be painful by itself.

File: static/docs/respecConfig.js (hunk 3 above)
Comment (sidvishnoi) [reply static/docs/respecConfig.js:126]: Still need to split `/docs/` at docs. I think this one is good.

**Pull request merged**